### PR TITLE
Enhance alvearie ingestion chart to include custom NiFi processors

### PR DESCRIPTION
**Pull request by luisgarcc**
This commit creates a Dockerfile to build the custom HL7 to FHIR and ACD
NiFi processors into a container that can then be used to load those
NARs into a NiFi server container.

It also creates an init container on the nifi chart that will copy
the alvearie custom NARs into a shared volume, that volume will be
mounted on the nifi server container and the `customLibPath` property
will be set to that volume so that nifi knows to load custom processors
from there.

### CODE REVIEW

**Review by ewill00 — APPROVED** [review]
LGTM

**Review by ewill00 — COMMENTED** [review]
Interesting comment I saw in the orchestration channel (from Dave Snyder)

_The nifi deployment has an option to configure a cloud object storage share which gets mounted to /opt/nifi/shared/files and /opt/nifi/shared/nar_extensions.  If you have this configured, anything put under the file path would be available to all nodes in the cluster.  This would be the best way to deliver library code like jar files, etc._

Curious how this relates to how we have done it.

File: nifi-processors/docker/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Alvearie NiFi Custom Processors Container
+
+This container includes the Alvearie Ingestion custom NiFi processors. It does not contain any executables, it is simply meant to be mounted as a custom processor directory in an NiFi installation.
+
+The container can be downloaded from Dockher Hub here: [luisgarc/alvearie-nars](https://hub.docker.com/repository/docker/luisgarc/alvearie-nars).
```
Comment (ibmatc) [inline nifi-processors/docker/README.md:5]: Should move this location away from luisgarc.

**Review by ibmatc — APPROVED** [review]
LGTM

**Comment by luisgarcc** [comment]
> 
> 
> Interesting comment I saw in the orchestration channel (from Dave Snyder)
> 
> _The nifi deployment has an option to configure a cloud object storage share which gets mounted to /opt/nifi/shared/files and /opt/nifi/shared/nar_extensions. If you have this configured, anything put under the file path would be available to all nodes in the cluster. This would be the best way to deliver library code like jar files, etc._
> 
> Curious how this relates to how we have done it.

I am not sure what _"the nifi deployment"_ means, can you clarify? Are you guys talking about some additional internal NiFi Chart used to deploy NiFi that has this cloud storage config? 

For this PR I am using the dedicated extension point that the official NiFi Chart has to add custom NARs.

File: nifi-processors/docker/README.md (hunk 1 above)
Comment (luisgarcc) [reply nifi-processors/docker/README.md:5]: Yeah agree. I think we need an Alvearie account in Docker Hub.

**Comment by ewill00** [comment]
I'm pretty sure it is referring to the CDT environment and overall 'orchestration' that goes beyond just NiFi and is considering muti-node deployments, etc.  But wanted to point out the pattern they are using for awareness.

**Pull request merged**